stdpopsim/slim_engine.py

```diff
@@ -293,13 +293,22 @@
 
     // Save/restore events. These should come before all other events.
     if (length(drawn_mutations) > 0) {
+        n_checkpoints = 0;
         for (i in 0:(ncol(drawn_mutations)-1)) {
             save = drawn_mutations[4,i] == 1;
             if (!save) {
                 next;
             }
-            g = G_start + gdiff(T_start, drawn_mutations[0,i]);
+
+            // Saving the state at more than one timepoint can can cause
+            // incorrect conditioning in the rejection samples.
+            if (n_checkpoints > 0) {
+                err("Attempt to save state at more than one checkpoint");
```

Comment:
Doesn't this just error if `drawn_mutations` has more than one column?

Reply:
oh I see, never mind!

Reply:
No, because of the "next" a few lines before. Multiple drawn mutations are tested here:

https://github.com/popsim-consortium/stdpopsim/blob/a0da48882a7eb69a77778a620e78447ef775f3ce/tests/test_slim_engine.py#L1378